fvm/scriptEnv.go

```diff
@@ -660,70 +660,43 @@ func (e *ScriptEnv) ImplementationDebugLog(message string) error {
 	return nil
 }
 
-func (e *ScriptEnv) ProgramParsed(location common.Location, duration time.Duration) {
+func (e *ScriptEnv) RecordTrace(operation string, location common.Location, duration time.Duration, logs []opentracing.LogRecord) {
 	if e.isTraceable() {
-		locStr := ""
+		if logs == nil {
+			logs = make([]opentracing.LogRecord, 0)
+		}
```

Comment:
Does this have to be an allocated slice? Can it just be `nil`, or does `RecordSpanFromParent` not like that? (Same for the other `RecordTrace` function below)

Reply:
actually it can be nil, mostly I added this to handle the logic in the if location != nil condition, but I could move it there, so we don't have do any allocation for empty locations

Reply:
move the logic inside the location logging part